pkg/resource/cluster.go

```diff
@@ -105,6 +144,90 @@ func (cluster Cluster) StatefulSetName() string {
 	return cluster.Name()
 }
 
+func (cluster Cluster) JobName() string {
+	return cluster.Name()
+}
+func (cluster Cluster) IsSupportedImage() bool {
+	image := cluster.GetCockroachDBImageName()
+	return !strings.EqualFold(image, NotSupportedVersion)
+}
+func (cluster Cluster) LookupSupportedVersion(version string) (string, bool) {
+	if version == "" {
+		return "", false
+	}
+	supportedVersions := getSupportedCrdbVersions()
+	for _, v := range supportedVersions {
+		if strings.EqualFold(version, v) {
+			return v, true
+		}
+	}
+	return "", false
+}
+
+func (cluster Cluster) GetAnnotationVersion() string {
+	return cluster.getAnnotation(CrdbVersion)
+}
+
+func (cluster Cluster) GetAnnotationContainerImage() string {
+	return cluster.getAnnotation(CrdbContainerImage)
+}
+
+func (cluster Cluster) GetAnnotationHistory() string {
+	return cluster.getAnnotation(CrdbHistory)
+}
+
+func (cluster Cluster) getAnnotation(key string) string {
+	if val, ok := cluster.cr.Annotations[key]; !ok {
+		return ""
+	} else {
+		return val
+	}
+}
+func (cluster Cluster) SetAnnotationVersion(version string) {
+	cluster.cr.Annotations[CrdbVersion] = version
+}
+func (cluster Cluster) SetAnnotationContainerImage(containerimage string) {
+	cluster.cr.Annotations[CrdbContainerImage] = containerimage
+}
+
+func (cluster Cluster) GetCockroachDBImageName() string {
```

Comment:
We need unit testing here, and also more error handling. Future work

Reply:
agree.. I will add an issue on this